KarhooUISDK/Screens/BookingScreen/KarhooBookingViewController.swift

```diff
@@ -376,23 +372,42 @@ public final class KarhooBookingScreenBuilder: BookingScreenBuilder {
         self.locationService = locationService
     }
     
-    public func buildBookingScreen(journeyInfo: JourneyInfo? = nil,
-                                   passengerDetails: PassengerDetails? = nil,
-                                   callback: ScreenResultCallback<BookingScreenResult>?) -> Screen {
+    public func buildBookingScreen(
+        journeyInfo: JourneyInfo? = nil,
+        passengerDetails: PassengerDetails? = nil,
+        callback: ScreenResultCallback<BookingScreenResult>?
+    ) -> Screen {
         PassengerInfo.shared.set(details: passengerDetails)
-
-        var validatedJourneyInfo: JourneyInfo?
-
-        if let date = journeyInfo?.date {
-            validatedJourneyInfo =  Date(timeIntervalSinceNow: 0).compare(date) ==
-                .orderedDescending ? nil : journeyInfo
-        } else {
-            validatedJourneyInfo = journeyInfo
+        
+        var validatedJourneyInfo: JourneyInfo? {
+            guard let origin = journeyInfo?.origin else {
+                return nil
+            }
+            
+            var isDateAllowed: Bool {
+                guard let injectedDate = journeyInfo?.date else {
+                    return false
+                }
+                return injectedDate >= Date()
```

Comment:
if you want to validate date the propper comparasion should be 'injectedDate >= [Date() + 1hour]